masonry/src/widget/grid.rs

```diff
@@ -0,0 +1,442 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use accesskit::Role;
+use smallvec::SmallVec;
+use tracing::{trace_span, Span};
+use vello::kurbo::{Affine, Line, Stroke};
+use vello::Scene;
+
+use crate::theme::get_debug_color;
+use crate::widget::WidgetMut;
+use crate::{
+    AccessCtx, AccessEvent, BoxConstraints, EventCtx, LayoutCtx, LifeCycle, LifeCycleCtx, PaintCtx,
+    Point, PointerEvent, Size, StatusChange, TextEvent, Widget, WidgetId, WidgetPod,
+};
+
+pub struct Grid {
+    children: Vec<Child>,
+    grid_width: i32,
+    grid_height: i32,
+    grid_spacing: f64,
+    //old_bc: BoxConstraints,
+    //needs_layout: bool,
```

Comment:
I would probably just remove the layout caching for now. Layout caching is going to be handled directly by Masonry eventually.

Reply:
Removed. I agree that letting masonry handle it is a better way forward.